backend/Techfesia2019/accounts/views.py

```diff
@@ -0,0 +1,164 @@
+# from django.shortcuts import render
+# from django.http import HttpResponse, JsonResponse
+from rest_framework.renderers import JSONRenderer
+from rest_framework.views import APIView
+from rest_framework import generics
+from registration.models import User
+from registration.serializers import UserSerializer
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from registration.permissions import IsStaffUserOrPost, IsStaffUser, IsStaffUserOrAuthenticated
+from rest_framework_simplejwt import tokens
+
+# Create your views here.
+
+
+# class UsersView(generics.ListAPIView):
+#     permission_classes = (IsStaffUser,)
+#     # limit = int(super().request.query_params['limit'])
+#     queryset = models.User.objects.filter(is_staff=False)
+#     serializer_class = UserSerializer
+
+
+class UsersViewPost(APIView):
+    permission_classes = (IsStaffUserOrPost,)
+
+    def get(self, request, format=None):
+        print(request.query_params)
+        limit, page = 3, 1
+        if request.query_params.get('limit') is not None:
+            limit = int(request.query_params.get('limit'))
+        if request.query_params.get('page') is not None:
+            page = int(request.query_params.get('page'))
+        users = User.objects.order_by('username')
+        if users.count()//limit + 1 < page:
+            data = {"message": "This page does not exist", "pageSize": limit, "noOfPages": page}
+        else:
+            users = users[(page-1)*limit, page*limit]
+            serializer = UserSerializer(users, many=True)
+            data = {'users': serializer.data, "currentPage": page, "noOfPages": users.count()//limit+1}
+
+        return Response(data)
+
+    def post(self, request, format=None):
+        print(request.data)
```

Comment:
I am not reviewing this for now as the task was only for models. Once we complete models and move on to serializers/views, remind me to check this again. I will review it then and integrate